### ISSUE TITLE
Globally installed CLI to check local node_modules for plugins and configs?

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#413]
Ref https://github.com/stylelint/stylelint/pull/412#issuecomment-148138344

Is this a blocker to our release? Feels like an enhancement we could do after this release as globally installing stylelint doesn't really get a mention anywhere in the README. Plus globally installing and extending is like a double-whammy.

**Comment by davidtheclark** [comment]
Well, it does seem like global installation will be taken for granted when a CLI is involved. I think we should consider this a blocker unless we determine that it is a huge problem -- in which case we might just clearly document the restriction.

**Comment by jeddy3** [comment]
SGTM :)

**Comment by davidtheclark** [comment]
@jeddy3 : I'd like to clarify the scenario here. Can you tell me more about the directory structure and where you ran the command? Maybe this isn't a problem ...

Let's say we have this structure:

- home
  - dev
    - project
      - .stylelintrc
      - test.css
      - node_modules
        - stylelint-config-suitcss

Case A. I think our goal is that the CLI should work if you run the following command *from `home/dev/project`*: `stylelint test.css`, or `stylelint *.css`.

Case B. However, I do not think it should work if you try to run that command, say, from `home` or `home/dev`. Because in that case it would not find the `.stylelintrc` file.

Doing that doesn't work in ESLint, either, I think (after a little experimentation).

So, @jeddy3, when you got your failure was it in Case A or Case B?

**Comment by davidtheclark** [comment]
Case C: It *should* work if you run from `home` the command `stylelint dev/project/test.css --config dev/project/.stylelintrc`.

Does that cover all the cases? Do you agree?

**Comment by davidtheclark** [comment]
Here's another case, and this one is killing me:

- home
  - dev
    - project
      - foo
        - myConfig.json
      - test.css
      - node_modules
        - stylelint-config-suitcss

I guess that from `home` they should be able to run `stylelint dev/project/test.css --config dev/project/foo/myConfig.json` and somehow we can find `stylelint-config-suitcss` given that. Ugh.

**Comment by jeddy3** [comment]
@davidtheclark Oops, I merged the PR before realising these messages were here. 

Also, sorry about not getting back to you while you were working on the issue, I was sleeping :)

That directory structure is a fair representation. I was originally just trying "Case A". But I've just tried B, C and the last one on your PR.

A, C and the last one work. Whereas B doesn't. I agree that's the expected behaviour here, so I think we're all done here :)

Thanks again for tackling this one, especially thinking up, and getting working, the additional two use cases!
